src/EditorFeatures/CSharpTest/Diagnostics/UseAutoProperty/UseAutoPropertyTests.cs

```diff
@@ -0,0 +1,223 @@
+@"class Class { int [|i|], j, k; int P { get { return i; } } }",
+@"class Class { int j, k; int P { get; } }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        public void TestFieldWithMultipleDeclarators2()
+        {
+            Test(
+@"class Class { int i, [|j|], k; int P { get { return j; } } }",
+@"class Class { int i, k; int P { get; } }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        public void TestFieldWithMultipleDeclarators3()
+        {
+            Test(
+@"class Class { int i, j, [|k|]; int P { get { return k; } } }",
+@"class Class { int i, j; int P { get; } }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        public void TestFieldAndPropertyInDifferentParts()
+        {
+            Test(
+@"partial class Class { [|int i|]; } partial class Class { int P { get { return i; } } }",
+@"partial class Class { } partial class Class { int P { get; } }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        public void TestNotWithFieldWithAttirbute()
+        {
+            TestMissing(
+@"class Class { [|[A]int i|]; int P { get { return i; } } }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        public void TestUpdateReferences()
+        {
+            Test(
+@"class Class { [|int i|]; int P { get { return i; } } public Class() { i = 1; } }",
+@"class Class { int P { get; } public Class() { P = 1; } }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        public void TestUpdateReferencesConflictResolution()
+        {
+            Test(
+@"class Class { [|int i|]; int P { get { return i; } } public Class(int P) { i = 1; } }",
+@"class Class { int P { get; } public Class(int P) { this.P = 1; } }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        public void TestWriteInConstructor()
+        {
+            Test(
+@"class Class { [|int i|]; int P { get { return i; } } public Class() { i = 1; } }",
+@"class Class { int P { get; } public Class() { P = 1; } }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        public void TestWriteInNotInConstructor1()
+        {
+            Test(
+@"class Class { [|int i|]; int P { get { return i; } } public Foo() { i = 1; } }",
+@"class Class { int P { get; set; } public Foo() { P = 1; } }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        public void TestWriteInNotInConstructor2()
+        {
+            Test(
+@"class Class { [|int i|]; public int P { get { return i; } } public Foo() { i = 1; } }",
+@"class Class { public int P { get; private set; } public Foo() { P = 1; } }");
+        }
+    }
```

Comment:
Add tests for trivia? are leading, trailing comments kept. Is the generated member simplified and formatted.